halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/services/v1/GenerateService.java

```diff
@@ -1,4 +1,103 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package com.netflix.spinnaker.halyard.config.services.v1;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.spinnaker.halyard.config.config.v1.FileWriter;
+import com.netflix.spinnaker.halyard.config.errors.v1.HalconfigException;
+import com.netflix.spinnaker.halyard.config.model.v1.node.Node;
+import com.netflix.spinnaker.halyard.config.model.v1.node.NodeFilter;
+import com.netflix.spinnaker.halyard.config.model.v1.node.NodeReference;
+import com.netflix.spinnaker.halyard.config.model.v1.problem.Problem;
+import com.netflix.spinnaker.halyard.config.model.v1.problem.ProblemBuilder;
+import com.netflix.spinnaker.halyard.config.spinnaker.v1.ComponentName;
+import com.netflix.spinnaker.halyard.config.spinnaker.v1.ComponentProfile;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Component
+@Slf4j
 public class GenerateService {
+  @Autowired
+  ComponentProfile componentProfile;
+
+  @Autowired
+  LookupService lookupService;
+
+  @Autowired
+  String spinnakerOutputPath;
+
+  @Autowired
+  FileWriter yamlWriter;
+
+  @Autowired
+  Yaml yamlParser;
+
+  @Autowired
+  ObjectMapper objectMapper;
+
+  public void generateConfig(NodeReference nodeReference) {
+    NodeFilter filter = NodeFilter.makeAcceptAllFilter()
```

Comment:
This is odd having just read "makeRejectAllFilter().extendReference()".

At first I thought it was wrong, but I think this is right.
One is narrowing the other is widening. The filter is not relevant to all the data which is why the two are different. But what is odd is that they are both extending.

Dont change anything, i'm just thinking "out loud". Rather than all or none filters you are creating a narrowing (initially all) or widening filter (initially none) then adding into that. When you add into a narrowing filter, it shrinks in scope. When you add into a widening filter, it grows in scope. The implementation is exactly what you are doing but the model is a bit different so the actions have different names and is hopefully clearer.

It's not so bad the way it is now, but AcceptAll.extend seems odd because how can I extend all? And RejectAll.extend seems odd because I think the emphasis is on Reject, not on All so "extend" sounds like shrink (extend the scope of rejection) in which case how can I shrink None.